src/LibraryManager.Vsix/UI/Models/InstallDialogViewModel.cs

```diff
@@ -372,29 +388,186 @@ private static bool IsAnyFileSelected(IReadOnlyList<PackageItem> children)
+                else
                 {
-                    _closeDialog(true);
-                });
+                    // libman.json file is open, so we will write to the textBuffer.
+                    _dispatcher.Invoke(() =>
+                    {
+                        _closeDialog(true);
+                    });
+
+                    InsertIntoTextBuffer(document, libraryInstallationState);
+
+                    // Save manifest file so we can restore library files.
+                    rdt.SaveFileIfDirty(Path.GetFullPath(_configFileName));
+                }
             }
             catch { }
         }
 
+        private void InsertIntoTextBuffer(IVsTextBuffer document, LibraryInstallationState libraryInstallationState)
+        {
+            ITextBuffer textBuffer = GetTextBuffer(document);
+
+            if (textBuffer != null)
+            {
+                string insertionText;
+
+                JSONMember libraries = GetLibraries(textBuffer);
+
+                if (libraries != null && libraries.Children != null)
+                {
+                    JSONArray jsonArray = libraries.Children.OfType<JSONArray>().First();
+
+                    if (jsonArray != null)
+                    {
+                        string newLibrary = GetLibraryToBeInserted(libraryInstallationState);
+                        bool containsLibrary = jsonArray.BlockItemChildren.Count() > 0;
+
+                        if (containsLibrary)
+                        {
+                            insertionText = _commaDelimiter + Environment.NewLine + newLibrary + Environment.NewLine;
+                        }
+                        else
+                        {
+                            insertionText = newLibrary + Environment.NewLine;
+                        }
+
+                        int insertionIndex = FindInsertionIndex(textBuffer);
+
+                        if (insertionIndex > 0)
+                        {
+                            FormatSelection(textBuffer, insertionIndex, insertionText);
+                        }
+                    }
+                }
+            }
+        }
+
+        private void FormatSelection(ITextBuffer textBuffer, int insertionIndex, string insertionText)
+        {
+            Controller.TextViewData viewData = Controller.TextViewConnectionListener.GetTextViewDataForBuffer(textBuffer);
+            ITextView textView = viewData.LastActiveView;
+
+            CompoundUndoAction compoundUndoAction = new CompoundUndoAction(textView, textBuffer, true);
+            compoundUndoAction.Open(Text.FormatUndoAction);
+
+            using (ITextEdit textEdit = textBuffer.CreateEdit())
+            {
+                textEdit.Insert(insertionIndex, insertionText);
+                textEdit.Apply();
+
+                IOleCommandTarget commandTarget = Shell.Package.GetGlobalService(typeof(Shell.Interop.SUIHostCommandDispatcher)) as IOleCommandTarget;
+
+                if (commandTarget != null)
```

Comment:
>if (commandTarget != null) [](start = 16, length = 26)

I could go either way, but there probably isn't really a need to check this point. If it's null, VS is toast.